apps/bluetooth/app.js

```diff
@@ -115,6 +115,9 @@ module.exports = function (activity) {
         if (wifi.getWifiState() === wifi.WIFI_CONNECTED) {
           return activity.tts.speak(text, { impatient: false }).catch((err) => {
             logger.error('play tts error: ', err)
+            if (alternativeVoice != null) {
+              return activity.playSound(alternativeVoice)
+            }
```

Comment:
Couldn't it check before speaking?

Reply:
Already checked no connection, is there an interface to check weak connection? such as `wifi.getWifoState() === wifi.WIFI_WEAK_CONNECTION` ?

Reply:
Not yet, I gotta resolve for now :)